### PR TITLE
Feat 7045 pdf

### PR DESCRIPTION
**Pull request by phpgeekfr**

### CODE REVIEW

File: libs/media/src/lib/pipes/storageFile.pipe.ts [hunk 1]
```diff
@@ -5,19 +5,23 @@ import { CollectionHoldingFile, FileLabel, getFileMetadata } from '../+state/sta
...
 import { Observable } from 'rxjs';
 
+export function toStorageFile(value: StorageFile | string, collection: CollectionHoldingFile, label: FileLabel, docId: string) {
+  if (typeof value === "string") {
+    const metadata = getFileMetadata(collection, label, docId)
+    return createStorageFile({
+      storagePath: value,
+      ...metadata
+    })
+  } else return value;
```
Comment (GrandSchtroumpf) [inline libs/media/src/lib/pipes/storageFile.pipe.ts:15]:
```suggestion
  if (typeof value !== "string") return value;
  const metadata = getFileMetadata(collection, label, docId)
  return createStorageFile({
    storagePath: value,
    ...metadata
  })
```

File: libs/utils/src/lib/pdf.service.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        return {
+          id: m.objectID,
+          posterUrl
+        };
+      }),
+      app
+    };
+
+    // @dev if using emulator, use `http://localhost:5001/${projectId}/${firebaseRegion}/createPdf`
+    const url = `https://${firebaseRegion}-${projectId}.cloudfunctions.net/createPdf`;
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/pdf.service.ts:34]: The code should work the same in both cases : 
```typescript
const url = useEmulators
  ? `http://localhost:5001/${projectId}/${firebaseRegion}/createPdf`
  : `https://${firebaseRegion}-${projectId}.cloudfunctions.net/createPdf`;
```

File: libs/utils/src/lib/pdf.service.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    await new Promise(resolve => {
+      this.http.post(url, params, { responseType: 'arraybuffer' })
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/pdf.service.ts:37]: use `fetch` if you want to use a promise; 
```typescript
fetch(`${url}?${params}`).then(res => res.blob()).then(...)
```

File: libs/utils/src/lib/static-model/static-model.ts [hunk 4]
```diff
@@ -288,9 +288,9 @@ export const movieFormatQuality = {
 
 export const movieLanguageTypes = {
   original: 'Original',
-  dubbed: 'Dubbings',
-  subtitle: 'Subtitles',
-  caption: 'Closed Captions',
+  dubbed: 'subs',
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/static-model/static-model.ts:291]: This change will be on every form. Is that ok ?

File: libs/utils/src/lib/utils.ts [hunk 5]
```diff
@@ -214,3 +215,31 @@ export function getWatermark(email: string = '', firstName: string = '', lastNam
 
   return `data:image/svg+xml;utf8,${encodeURIComponent(svg)}`;
 }
+
+export function toLabel(value: string | string[], scope: Scope, joinWith? : string, endWith?: string): string {
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/utils.ts:219]: I thought we already had a function `toLabel`...

File: apps/backend-functions/src/createPdf.ts [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+  const args = ['--no-sandbox', '--disable-setuid-sandbox'];
+  const browser = await puppeteer.launch({ args: args });
+  const page = await browser.newPage();
+
+  // Wait for the page to load completely
+  await page.setContent(html, { waitUntil: 'networkidle0' });
+  await page.evaluateHandle('document.fonts.ready');
+
+
+  const cssHeader = [];
```
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply apps/backend-functions/src/createPdf.ts:162]: Headers and footers are added by `puppeteer` as separate html files. So the css in the html cannot be used

File: libs/utils/src/lib/static-model/static-model.ts (hunk 4 above)
Comment (phpgeekfr) [reply libs/utils/src/lib/static-model/static-model.ts:291]: I need to confirm with @2amarkina but yes, this is the wording that professionals uses and understands

Comment (2amarkina) [reply libs/utils/src/lib/static-model/static-model.ts:291]: It's ok for business team to use the following short versions everywhere:

File: libs/utils/src/lib/utils.ts (hunk 5 above)
Comment (phpgeekfr) [reply libs/utils/src/lib/utils.ts:219]: This is the one and only. I moved it so I can import it in backend without importing angular related stuff

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**